src/MergeContext.js

```diff
@@ -499,58 +588,73 @@ class MergeContext {
-        // the branch has a single required check named 'Jenkins(build test)',
-        // the bot will wait for three checks with 'Jenkins(build test).*' names.
-        const needMatching = checksNumber !== undefined;
-        const requiredChecksNumber = needMatching ? checksNumber : requiredContexts.length;
-
-        // An array of [{context, state}] elements
-        let requiredChecks = [];
+            return new StatusResult(0, "PR");
+        }
+
+        let statusResult = new StatusResult(requiredContexts.length, "PR");
         // filter out non-required checks
         for (let st of combinedStatus.statuses) {
-            if (requiredContexts.find((el) => { return needMatching ?
-                         st.context.startsWith(el.trim()) : (el.trim() === st.context.trim());}))
-                requiredChecks.push({context: st.context, state: st.state, targetUrl: st.target_url, description: st.description});
-        }
-
-        if (requiredChecks.length < requiredChecksNumber || requiredChecks.find(check => check.state === 'pending'))
-            return 'pending';
-
-        const prevLen = requiredChecks.length;
-        requiredChecks = requiredChecks.filter(check => check.state === 'success');
-        const ret = (prevLen === requiredChecks.length);
-        if (ret && andSupplyRequired) {
-            for (let requiredContext of requiredContexts) {
-                // go further only if the passed check context is not a required one
-                if (requiredChecks.find(el => el.context.trim() === requiredContext.trim()) === undefined) {
-                    // go further only if the passed check context matches the required one
-                    const matched = requiredChecks.find(el => el.context.startsWith(requiredContext.trim()));
-                    // Before a 'staged' commit can be applied, it should have all required checks passed.
-                    // We create a new "required" check, taking other attributes like targetUrl
-                    // from an already succeeded (matching) check (there can be several matching checks,
-                    // e.g., from different Jenkins nodes). After that, there will be two checks
-                    // referencing the same targetUrl.
-                    if (matched && !this._dryRun("required status check creation"))
-                        await GH.createStatus(ref, "success", matched.targetUrl, matched.description, requiredContext);
+            if (requiredContexts.find(el => el.trim() === st.context.trim()) !== undefined)
+                statusResult.addStatus(new StatusCheck(st.context, st.state, st.target_url, st.description));
+        }
+        return statusResult;
+    }
+
+    // returns filled StatusResult object
+    async __checkStagingStatuses() {
+        let combinedStatus = await GH.getStatuses(this._tagSha);
+        // assert(combinedStatus.statuses.length <= Config.stagingChecks());
```

Comment:
Unfortunately, this assert cannot work here (checking that no new staging checks exceeding our 'stagingChecks()' parameter appeared). The reason is that for guarded_run mode we get here again, already after _supplyStagingWithPrRequired() created extra checks (PR checks for staged commit).